clang-tools-extra/test/clang-tidy/checkers/readability/redundant-casting.cpp

```diff
@@ -221,3 +221,16 @@ void testRedundantDependentNTTPCasting() {
   // CHECK-MESSAGES: :[[@LINE-4]]:25: note: source type originates from referencing this non-type template parameter
   // CHECK-FIXES: {{^}}  T a = V;
 }
+
+enum E1 : char {};
+enum class E2 : char {};
+
+void testEnum(E1 e1, E2 e2){
+  char a = static_cast<char>(e1);
```

Comment:
That isn't a redundant cast. Such casts should be done in explicit way. In C++ is even better to use:
std::to_underlying since C++23 or std::underlying_type now.